js/blockedLinks.js

```diff
@@ -0,0 +1,106 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+/* globals gPinnedLinks, gUserDatabase, async*/
+"use strict";
+
+/**
+ * Singleton that keeps track of all blocked links in the grid.
+ */
+(function(exports) {
+  const gBlockedLinks = {
+    /**
+     * The cached list of blocked links.
+     */
+    _links: {},
+
+
+    /**
+     * Load the blocked links from gUserDatabase and cache them.
+     */
+    initBlockedLinks() {
+      var loadPromise = gUserDatabase.load("prefs", "blockedLinks");
+      loadPromise.then(loadedLinks => {
+        this._links = (loadedLinks && loadedLinks.length) ? JSON.parse(loadedLinks) : {};
+      });
+      return loadPromise;
+    },
+
+    /**
+     * The list of blocked links.
+     */
+    get links() {
+      return this._links;
```

Comment:
good point. I think we may not need this getter at all. The only time it's used is here: https://dxr.mozilla.org/mozilla-central/source/toolkit/modules/NewTabUtils.jsm#1315

and in that case we only need to return the number of blocked links. We can add that functionality once we move over this `_collect()` function from Firefox